cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicControllerConfig.java

```diff
@@ -0,0 +1,282 @@
+     */
+    public TopicControllerConfig withNamespace(String namespace) {
+        this.namespace = namespace;
+        return this;
+    }
+
+    /**
+     * Specify the Docker image to use for the Topic Controller
+     *
+     * @param image the Docker image to use for the Topic Controller
+     * @return current TopicControllerConfig instance
+     */
+    public TopicControllerConfig withImage(String image) {
+        this.image = image;
+        return this;
+    }
+
+    /**
+     * Specify the interval between periodic reconciliations
+     *
+     * @param reconciliationInterval the interval between periodic reconciliations
+     * @return current TopicControllerConfig instance
+     */
+    public TopicControllerConfig withReconciliationInterval(String reconciliationInterval) {
+        this.reconciliationInterval = reconciliationInterval;
+        return this;
+    }
+
+    /**
+     * Specify the Zookeeper session timeout
+     *
+     * @param zookeeperSessionTimeout the Zookeeper session timeout
+     * @return current TopicControllerConfig instance
+     */
+    public TopicControllerConfig withZookeeperSessionTimeout(String zookeeperSessionTimeout) {
+        this.zookeeperSessionTimeout = zookeeperSessionTimeout;
+        return this;
+    }
+
+    /**
+     * Returns a TopicControllerConfig instance from a corresponding JSON representation
+     *
+     * @param json  topic controller configuration JSON representation
+     * @return  TopicControllerConfig instance
+     */
+    public static TopicControllerConfig fromJson(TopicControllerConfig config, JsonObject json) {
+
+        String image = json.getString(TopicControllerConfig.IMAGE_FIELD);
+        if (image != null) {
+            config.withImage(image);
+        }
+
+        String namespace = json.getString(TopicControllerConfig.NAMESPACE_FIELD);
+        if (namespace != null) {
+            config.withNamespace(namespace);
+        }
+
+        String reconciliationInterval = json.getString(TopicControllerConfig.RECONCILIATION_INTERVAL_FIELD);
+        if (reconciliationInterval != null) {
+            // TODO : add parsing and validation
+            config.withReconciliationInterval(reconciliationInterval);
+        }
+
+        String zookeeperSessionTimeout = json.getString(TopicControllerConfig.ZOOKEEPER_SESSION_TIMEOUT_FIELD);
+        if (zookeeperSessionTimeout != null) {
+            // TODO : add parsing and validation
+            config.withZookeeperSessionTimeout(zookeeperSessionTimeout);
+        }
+
+        return config;
+    }
+
+    /**
+     * Compute the difference between two TopicControllerConfig instances
+     *
+     * @param other the other instance to compare with
+     * @return the result with all differences
+     */
+    public TopicControllerConfigResult diff(TopicControllerConfig other) {
+
+        TopicControllerConfigResult diffResult = new TopicControllerConfigResult();
+
+        diffResult
```

Comment:
Why do we need `TopicControllerConfigResult` with these 4 flags when (AFAICS) we just || those flags together any way in `TopicController.diff()`. Did I miss something?

Reply:
I got your point. I'm following the pattern we already have in all the other cases and maybe in the future it could be useful having the specific reason why there is a difference.

Reply:
Do we really need it in those other places?